src/test/java/com/revaturelabs/ask/AskApplicationModelTests.java

```diff
@@ -0,0 +1,614 @@
+
+    User testUserDiffResponses = new User();
+    testUserDiffResponses.setId(10);
+    testUserDiffResponses.setUsername("username");
+    testUserDiffResponses.setPassword("password");
+    testUserDiffResponses.setExpert(true);
+    testUserDiffResponses.setExpertTags(tags);
+    testUserDiffResponses.setQuestions(questions);
+    Set<Response> responses2 = new HashSet<Response>();
+    responses2.add(testResponse2);
+    testUserDiffResponses.setResponses(responses2);
+
+    User testUserNullResponses = new User();
+    testUserNullResponses.setId(10);
+    testUserNullResponses.setUsername("username");
+    testUserNullResponses.setPassword("password");
+    testUserNullResponses.setExpert(true);
+    testUserNullResponses.setExpertTags(tags);
+    testUserNullResponses.setQuestions(questions);
+    testUserNullResponses.setResponses(null);
+
+    assertNotEquals(testUser.getExpertTags(), testUserAddTag.getExpertTags());
+    assertNotEquals(testUser.getUsername(), "Not the username");
+    assertNotEquals(testUserNullUsername.getUsername(), "Not the username");
+    assertEquals(testUserNullUsername.getUsername(), null);
+
+    // .hashcode()
+    assertTrue(testUser.hashCode() == testUserSame.hashCode());
+    assertFalse(testUser.hashCode() == testUserNullFields.hashCode());
+
+    // .equals()
+    assertTrue(testUser.equals(testUserSame));
+    assertTrue(testUserNullFields.equals(testUserNullFieldsSame));
+    assertFalse(testUser.equals(null));
+    assertFalse(testUser.equals(testImage1));
+    assertFalse(testUserDiffId.equals(testUserNullFields) & testUserDiffId.equals(testUser));
+    assertFalse(
+        testUserDiffUsername.equals(testUserNullFields) & testUserDiffUsername.equals(testUser));
+    assertFalse(
+        testUserNullUsername.equals(testUserNullFields) & testUserNullUsername.equals(testUser)
+            & testUserNullUsername.equals(testUserDiffUsername));
+    assertFalse(
+        testUserDiffPassword.equals(testUserNullFields) & testUserDiffPassword.equals(testUser));
+    assertFalse(
+        testUserNullPassword.equals(testUserNullFields) & testUserNullPassword.equals(testUser));
+    assertFalse(
+        testUserDiffExpert.equals(testUserNullFields) & testUserDiffExpert.equals(testUser));
+    assertFalse(testUserDiffTags.equals(testUserNullFields) & testUserDiffTags.equals(testUser));
+    assertFalse(testUserNullTags.equals(testUserNullFields) & testUserNullTags.equals(testUser));
+    assertFalse(
+        testUserDiffQuestions.equals(testUserNullFields) & testUserDiffQuestions.equals(testUser));
+    assertFalse(
+        testUserNullQuestions.equals(testUserNullFields) & testUserNullQuestions.equals(testUser));
+    assertFalse(
+        testUserDiffResponses.equals(testUserNullFields) & testUserDiffResponses.equals(testUser));
+    assertFalse(
+        testUserNullResponses.equals(testUserNullFields) & testUserNullResponses.equals(testUser));
+  }
```

Comment:
This test suite with the liberal use of static references was written during a previous sprint. It needs to be refactored.